presto-orc/src/test/java/com/facebook/presto/orc/TestOrcWriter.java

```diff
@@ -0,0 +1,122 @@
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+import io.airlift.units.DataSize;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static com.facebook.presto.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static com.facebook.presto.orc.OrcWriter.createOrcWriter;
+import static com.facebook.presto.orc.TestingOrcPredicate.ORC_ROW_GROUP_SIZE;
+import static com.facebook.presto.orc.TestingOrcPredicate.ORC_STRIPE_SIZE;
+import static com.facebook.presto.orc.metadata.CompressionKind.NONE;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Verify.verify;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+
+public class TestOrcWriter
+{
+    @Test
+    public void testWriteOutputStreamsInOrder()
+            throws IOException
+    {
+        ImmutableMap.Builder<String, String> metadata = ImmutableMap.builder();
+        metadata.put("columns", "test");
+        metadata.put("columns.types", "varchar");
+
+        // Ignore MAGIC and 5 indexes at the beginning; ignore metadata + footer + post script at the end
+        SliceOutput output = new TestStreamOrderingOutput(1 + 5, 1 + 1 + 1);
+        OrcWriter writer = createOrcWriter(
+                    output,
+                    ImmutableList.of("test1", "test2", "test3", "test4", "test5"),
+                    ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, VARCHAR, VARCHAR),
+                    NONE,
+                    new DataSize(128, MEGABYTE),
+                    ORC_STRIPE_SIZE,
+                    ORC_STRIPE_SIZE,
+                    ORC_ROW_GROUP_SIZE,
+                    new DataSize(32, MEGABYTE),
+                    ImmutableMap.of(),
+                    HIVE_STORAGE_TIME_ZONE,
+                    true);
+
+        String[] data = new String[]{"a", "bbbbb", "ccc", "dd", "eeee"};
+        Block[] blocks = new Block[data.length];
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 1024);
+        for (int i = 0; i < data.length; i++) {
+            byte[] bytes = data[i].getBytes();
+            for (int j = 0; j < 1024; j++) {
+                // force to write different data
+                bytes[0] = (byte) ((bytes[0] + 1) % 128);
+                blockBuilder.writeBytes(Slices.wrappedBuffer(bytes, 0, bytes.length), 0, bytes.length);
+                blockBuilder.closeEntry();
+            }
+            blocks[i] = blockBuilder.build();
+            blockBuilder = blockBuilder.newBlockBuilderLike(new BlockBuilderStatus());
+        }
+
+        writer.write(new Page(blocks));
+        writer.close();
+    }
+
+    private final class TestStreamOrderingOutput
```

Comment:
Instead of reverse engineering the output stream, how about we just read the stripe footer and compare the Stream data.  I'm thinking we use something like this: 
```java
Footer footer = new OrcReader(...).getFooter();

readStripeFooter(footer.getStripes().get(0), footer.getTypes(), new OrcMetadataReader(), ...);

   // this is a hacked version of com.facebook.presto.orc.StripeReader#readStripeFooter
    public static StripeFooter readStripeFooter(StripeInformation stripe, List<OrcType> types, MetadataReader metadataReader, OrcDataSource orcDataSource, Optional<OrcDecompressor> decompressor, AbstractAggregatedMemoryContext systemMemoryUsage)
            throws IOException
    {
        long offset = stripe.getOffset() + stripe.getIndexLength() + stripe.getDataLength();
        int tailLength = toIntExact(stripe.getFooterLength());

        // read the footer
        byte[] tailBuffer = new byte[tailLength];
        orcDataSource.readFully(offset, tailBuffer);
        try (InputStream inputStream = new OrcInputStream(orcDataSource.getId(), Slices.wrappedBuffer(tailBuffer).getInput(), decompressor, systemMemoryUsage)) {
            return metadataReader.readStripeFooter(types, inputStream);
        }
    }
```